### DISCUSSION TITLE
Setup guide for new user

### DISCUSSION THREAD

**Discussion by tomdd1976** [OpenRailAssociation/osrd#4398] [Q&A]
Thanks for this really interesting looking project. As a new user keen to test this out I'm a bit stuck on the installation instructions. Can someone help me with this?

The webpage shows the follwing instructions to get up and running:
```
# build and run the entire stack
docker-compose up -d --build

# generate and load an example infrastructure
poetry --directory=python/railjson_generator shell
./scripts/generate-infra.sh small_infra

# open the web app
xdg-open http://localhost:3000/

```

The "docker-compose up -d --build" completes and the front end can be reached from the localhost after I do this, so that's running at least. But I cannot successfully run the script to completion (either no python is found, the docker command is unknown or similar depending on where I am trying this from)

1. Firstly, should I use the Dev branch or the release 0.1.2? (I've tried both, but in the end I have the same problem below).
1. Are there any enviroment variables or similar that I need to set before building the containers? (e.g. setting OSRD_GIT_DESCRIBE to the repo path, etc.?). 
1. As a total Docker newbie, I'm a bit confused where do I run these commands? The python enviroment seems to be in the osrd-api container, so do I run Poetry and the generate-infra script from a terminal here? Or from some other (external?) location or container? Also, the generate-infra.sh script ends with "docker cp / docker exec..." commands, which don't seem to be runnable from inside a container.

I'm running Windows 10, but I got equally stuck trying with Ubuntu 22.04. It's just not clear to me how the system fits together.

Any hints and tips about what I should do would be appreciated.

Best,
tom_dd.

**Comment by younesschrifi** [comment]
Hello @tomdd1976,
 First of all, I'm sorry for the late reply. Thank you for your interest in our project! I can arrange a video conference to help you install osrd on your machine if that suits you. 
Write to me at: youness.chrifi@gmail.com and we can organise a google meet when you're available to help you install the application! 
Best, 
Youness
